src/components/Header/Header.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from "react";
+import Navbar from "react-bootstrap/Navbar";
+import Button from "react-bootstrap/Button";
+import Nav from "react-bootstrap/Nav";
+// import Form from "react-bootstrap/Form";
+// import FormControl from "react-bootstrap/FormControl";
+import Image from "react-bootstrap/Image";
+import logo from "../../assets/logo.png";
+import { Link } from "react-router-dom";
+import "./header.css";
+import { AUTH_TOKEN } from "../../constants";
+import { withRouter, RouteComponentProps } from "react-router-dom";
+
+interface Props extends RouteComponentProps<any> {}
+interface State {}
+
+class Header extends Component<Props, State> {
+  state = {};
+
+  render() {
+    const authToken = localStorage.getItem(AUTH_TOKEN);
+    console.log(authToken);
+    return (
+      <div className="header">
+        <Navbar bg="success" variant="dark">
+          <Navbar.Brand as={Link} to="/">
+            <Image src={logo} roundedCircle className="brandImage" />
+          </Navbar.Brand>
+          <Nav className="mr-auto">
+            {authToken == null && (
+              <Nav.Link as={Link} to="/">
+                Home
+              </Nav.Link>
+            )}
+            {authToken == null && (
+              <Nav.Link as={Link} to="/signup">
+                Signup
+              </Nav.Link>
+            )}
+            {authToken == null && (
+              <Nav.Link as={Link} to="/login">
+                Login
+              </Nav.Link>
+            )}
```

Comment:
Why check 3 times for each link? You can wrap them all in a React Fragment (https://reactjs.org/docs/fragments.html) so it doesn't mess with your styles but you can return the 3 of them together, like: 
```
{authToken == null && (
<>
              <Nav.Link as={Link} to="/">
                Home
              </Nav.Link>
...more Nav.Links
</>
            )}
            

```

Reply:
You are right.